exporters/committime/collector_base.py

```diff
@@ -251,6 +279,85 @@ def get_repo_from_jenkins(self, jenkins_builds):
                 )
         # If no repo is found, we will return None, which will be handled later on
 
+    def get_metric_from_pipelineruns(self, pipelines, namespace):
+        """Expects a sorted array of build data sorted by app label"""
+        metrics = []
+        for pipeline in pipelines:
+            runs = pipelines[pipeline]
+
+            for run in runs:
+                repo_url = 'http://default'
+                commit_sha = 'default'
+                image_hash = 'default'
+                image_location = 'http://default'
+                try:
+                    # metric = self.get_metric_from_build(build, app, namespace, repo_url)
+                    metric = CommitMetric(pipeline)
+                    # By convention need to get image location from PARAMS
+                    params = run.status.pipelineSpec.params
+                    if params:
+                        for param in params:
+                            if param.name == "IMAGE_NAME":
+                                image_location = param.default
+                    # save each taskRun inside the pipeline
+                    task_runs = run.status.taskRuns
+                    if task_runs:
+                        for taskrun in task_runs:
+                            if taskrun:
+                                # taskResults come in list inside array format for some reason
```

Comment:
@jlozano-rh Can you show me an example? [Their code](https://pkg.go.dev/github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1#PipelineRunStatusFields) implies it ought to be a map.
([they unfortunately don't have api reference docs yet](https://github.com/tektoncd/pipeline/issues/1250))

Reply:
@KevinMGranger yes, here a snippet of YAML code. But inside the committime exporter the object has a JSON format instead:

```
apiVersion: tekton.dev/v1beta1
kind: PipelineRun
metadata:
  name: myapp-ci-run-3bp0d5
  namespace: myapp-dev
spec:
  params:
  ...
status:
  ...
  taskRuns:
    myapp-ci-3bp0d5-build-and-push-mx4d2:
      pipelineTaskName: build-and-push
      status:
        ...
        taskResults:
          - name: IMAGE_DIGEST
            value: >-
              sha256:771f405494934dfeb46651b447f64f314fc74bcc495648f42794eb4934c890a7
        ...
    myapp-ci-3bp0d5-fetch-repository-hmnp5:
      pipelineTaskName: fetch-repository
      status:
        ...
        taskResults:
          - name: commit
            value: abdb7daef16279ff72d392be1b8ceccb5f121562
          - name: url
            value: 'https://github.com/myproject/myapp.git'
        ...
    myapp-ci-3bp0d5-get-app-version-hzfr6:
      pipelineTaskName: get-app-version
      status:
        ...
        taskResults:
          - name: value
            value: 0.1.1
        ...
    ...

```

Reply:
I'm not sure what you mean by json format?

Reply:
@KevinMGranger in the python code, if you print() the `run` object is shown as a JSON text. But is the same as YAML that you see in the OpenShift web console

Reply:
I looked into it-- I think it's the weirdness of `openshift.dynamic.ResourceField`. I'll do some testing to verify that.

Reply:
@KevinMGranger exactly, at the time I investigated and found no other way to do it.